### PR TITLE
Ratings support for trakt_list

### PR DESCRIPTION
**Pull request by brtkrbzhnv**
Support for getting all shows/episodes/movies with a particular rating.

### CODE REVIEW

**Comment by gazpachoking** [comment]
Should we allow for the all ratings list too? Maybe just make that another text option 'ratings'

**Comment by brtkrbzhnv** [comment]
Why not, I guess. I wouldn't use it, but maybe someone else would. Anyway, it would be great if we could get this merged.

**Comment by stevezau** [comment]
@gazpachoking any reason you didn't merge this? seems ok to me?

**Comment by snickers2k** [comment]
@brtkrbzhnv this looks great 
@gazpachoking please merge :)

**Comment by snickers2k** [comment]
okay.. i just tested this and becoming the following error:

2015-09-01 02:50 CRITICAL manager                       mapping values are not allowed here in "<unicode string>", line 230, column 19: list: rating: 10 ^

**Comment by OmgOhnoes** [comment]
I've been playing around with this and would it not be better to make it a separate setting instead of using the 'list' setting?

I'm not familiar with the trakt api (i tried but the documentation is lacklustre) but i would think it would be possible.

People use lists to manage their various shows, it would provide a lot of extra functionality to be able to filter by list AND rating instead of your global ratings.

For instance, i like a lot of shows and rate them 9, but maybe i only want the list of shows in X list that are rated 9 so i can set the quality to a higher level than shows rated 9 in other lists.

File: flexget/plugins/input/trakt_list.py [hunk 1]
```diff
@@ -20,31 +20,32 @@
...
-        # TODO: Should this have series_name? We normally only have that for episodes
+        'title': lambda i: '%s (%s)' % (i['show']['title'], i['show']['year']),
+        'series_name': lambda i: '%s (%s)' % (i['show']['title'], i['show']['year']),
         'imdb_id': 'show.ids.imdb',
         'tvdb_id': 'show.ids.tvdb',
         'tvrage_id': 'show.ids.tvrage',
         'tmdb_id': 'show.ids.tmdb',
-        'trakt_id': 'show.ids.trakt',
+        'trakt_show_id': 'show.ids.trakt',
         'trakt_slug': 'show.ids.slug'
```
Comment (OmgOhnoes) [inline flexget/plugins/input/trakt_list.py]: Could we add ```'trakt_user_rating': 'rating'``` to this? That way we can filter lists by rating via crossmatch.

File: flexget/plugins/input/trakt_list.py [hunk 2]
```diff
@@ -20,31 +20,32 @@
...
-        'tvdb_id': 'episode.ids.tvdb',
-        'tvrage_id': 'episode.ids.tvrage',
-        'trakt_id': 'show.ids.trakt',
-        'trakt_slug': 'show.ids.slug'
+        'imdb_id': 'show.ids.imdb',
+        'tvdb_id': 'show.ids.tvdb',
+        'tvrage_id': 'show.ids.tvrage',
+        'trakt_episode_id': 'episode.ids.trakt',
+        'trakt_show_id': 'show.ids.trakt',
+        'trakt_show_slug': 'show.ids.slug'
```
Comment (OmgOhnoes) [inline flexget/plugins/input/trakt_list.py]: Same thing here: ```'trakt_user_ep_rating': 'rating'```

**Comment by snickers2k** [comment]
@OmgOhnoes great thoughts

Can somebody please take a quick look? I think it's almost done.. just needs some adjustments?

**Comment by gazpachoking** [comment]
Merged in c7251688e6c8779dac5cbca2b34a13bc7338dbe4 @brtkrbzhnv Still around and want to update the wiki?

**Comment by snickers2k** [comment]
like i said, it's not working.
when using

list: rating: 10

following error: mapping values are not allowed here in "<unicode string>", line 419, column 19: list: rating: 10 ^

when using 

list: ratings
rating: 10

"rating" is not an allowed value or something...

**Comment by gazpachoking** [comment]
I think it should be:
```
list:
  rating: 10
```

**Comment by snickers2k** [comment]
Hey, it's been quite here for a while. Is this still working? Because i can't find it in the wiki and it seems to be not in the trakt_list.py anymore?!

Could you please add some ability to interact with ratings, no matter for which database (imdb/trakt/tvdb)? I tried it with plex_userrating, but since plex isn't supported with Plex-Home it's not possible for me.
It could be really awesome to rate a movie with 10 Stars to load it in a higher quality or delete it with 1 Stars or something... 

thanks

**Comment by gazpachoking** [comment]
It looks like we lost this feature when updating this to the new list style plugins. I think adding user rating support to trakt_lookup would be pretty nice, potentially nicer than being able to get lists of rated things with trakt_list.

**Comment by snickers2k** [comment]
i would be fine with any solution :) but of course, trakt_lookup would be the better way.

**Comment by cvium** [comment]
Should be available in the next release.

**Comment by snickers2k** [comment]
this is awesome! thank you so much!

**Comment by snickers2k** [comment]
is this already implemented? because it's not working for me.

>     if:
>      - trakt_movie_user_rating > '9': accept

**Comment by cvium** [comment]
You're going to have to be more specific

**Comment by snickers2k** [comment]
with this task:

`
  Update_Movie_Favorites:
    priority: 1
    template:
      - no_seen
      - no_global
    trakt_lookup: yes
    trakt_list:
      account: '{{ secrets.trakt.acc }}'
      list: collection
      type: movies
    crossmatch:
      from:
        - trakt_list:
            account: '{{ secrets.trakt.acc }}'
            list: '{{ secrets.trakt.Favorite_List }}'
            type: movies
      fields:
        - imdb_id
      action: reject
    if:
      - trakt_movie_user_rating > 9: accept
    list_add:
      - trakt_list:
          account: '{{ secrets.trakt.acc }}'
          list: '{{ secrets.trakt.Favorite_List }}'
          type: movies
`

it should accept a few dozens of movies rated above 9.. but instead

`2016-11-06 20:39 VERBOSE  details       Update_Movie_Favorites Summary - Accepted: 0 (Rejected: 0 Undecided: 1035 Failed: 0)
` 

the other way around; accept_all with "if below 9 -> reject" just accepts everything.

**Comment by cvium** [comment]
Use triple backticks for code blocks
\`\`\`
test
  test2
\`\`\`

```
test
  test2
```

Anyway, you're using it wrong. `trakt_lookup: yes` can't get any user specific data. You need to specify account.

**Comment by snickers2k** [comment]
oh, i overlooked that. thanks.

**Pull request closed without merging**